app/assets/javascripts/app/models/post/interactions.js

```diff
@@ -103,23 +103,21 @@ app.models.Post.Interactions = Backbone.Model.extend({
       , reshare = this.post.reshare()
       , flash = new Diaspora.Widgets.FlashMessages();
 
-    reshare.save({}, {
-      success : function(){
+    reshare.save()
+      .done(function(){
         flash.render({
           success: true,
           notice: Diaspora.I18n.t("reshares.successful")
         });
-      },
-      error: function(){
+        interactions.reshares.add(reshare);
+        if (app.stream) {app.stream.addNow(reshare)};
```

Comment:
Oh, okay, now I remember that I was having some problems with atom editor and those editing `rebase` before... I will fix this.